src/adapters/MeetingsJSONAdapter.js

```diff
@@ -479,494 +406,139 @@ export default class MeetingsJSONAdapter extends MeetingsAdapter {
-      meeting.localShare = await this.getDisplayStream();
-      meeting.remoteShare = null;
+    await this.updateMeeting(ID, async (meeting) => {
+      const meetingUpdated = {...meeting};
 
       if (meeting.localShare) {
-        // Handle browser's built-in stop Button
-        meeting.localShare.getVideoTracks()[0].onended = () => {
-          meeting.localShare = null;
-          document.dispatchEvent(new CustomEvent(SHARE_CONTROL, {detail: meeting}));
-        };
-      }
-    }
-
-    document.dispatchEvent(new CustomEvent(SHARE_CONTROL, {detail: meeting}));
-  }
-
-  /**
-   * Returns an observable that emits the display data of the share screen control.
-   *
-   * @param {string} ID ID of the meeting for which to update display
-   * @returns {Observable.<MeetingControlDisplay>} Observable that emits control display data
-   * @private
-   */
-  shareControl(ID) {
-    const inactive = {
-      ID: SHARE_CONTROL,
-      icon: 'share-screen-presence-stroke_26',
-      tooltip: 'Start Sharing',
-      state: MeetingControlState.INACTIVE,
-    };
-    const active = {
-      ID: SHARE_CONTROL,
-      icon: 'share-screen-presence-stroke_26',
-      tooltip: 'Stop Sharing',
-      state: MeetingControlState.ACTIVE,
-    };
-
-    const default$ = Observable.create((observer) => {
-      const meeting = this.datasource[ID];
-
-      if (meeting) {
-        observer.next(meeting.localShare ? active : inactive);
+        meetingUpdated.localShare.getTracks()[0].stop();
+        meetingUpdated.localShare = null;
       } else {
-        observer.error(new Error(`Could not find meeting with ID "${ID}"`));
+        meetingUpdated.localShare = await this.getDisplayStream();
+        meetingUpdated.remoteShare = null;
+
+        if (meeting.localShare) {
+          // Handle browser's built-in stop Button
+          meetingUpdated.localShare.getVideoTracks()[0].onended = () => {
+            meetingUpdated.localShare = null;
+          };
+        }
       }
 
-      observer.complete();
+      return meetingUpdated;
     });
-
-    const shareEvent$ = fromEvent(document, SHARE_CONTROL).pipe(
-      filter((event) => event.detail.ID === ID),
-      map((event) => (event.detail.localShare ? active : inactive)),
-    );
-
-    return concat(default$, shareEvent$);
   }
 
   /**
    * Handles the toggle of the roster.
    *
-   * @param {string} ID ID of the meeting for which to update display
+   * @param {string} ID  Id of the meeting for which to update display
    * @private
    */
   async toggleRoster(ID) {
-    const meeting = this.datasource[ID];
-
-    meeting.showRoster = !meeting.showRoster;
-
-    document.dispatchEvent(new CustomEvent(ROSTER_CONTROL, {detail: meeting}));
+    await this.updateMeeting(ID, async (meeting) => ({
+      ...meeting,
+      showRoster: !meeting.showRoster,
+    }));
   }
 
   /**
-   * Returns an observable that emits the display data of a roster control.
+   * Handles the toggle of the settings.
```

Comment:
actually, these are no longer "contol handlers", like they were before, they are "togglers".

So a better description would be "Toggles the settings display flag of a meeting"